File: VRP.Dtos/DataSetModel.cs
using System;
using System.Collections.Generic;
using VRP.Core.Interfaces;

namespace VRP.Dtos {
    public class DataSetModel : IId<long>, IDto, IExpirable {
        public long Id { get; set; }
        public DateTime InsertDate { get; set; }
        public DateTime? DeleteDate { get; set; }
        public string Name { get; set; }
        public string Logo { get; set; }
        public string Description { get; set; }
        public long UserId { get; set; }
        public ICollection<DataSetItemModel> DataSetItems { get; set; }
    }
}

File: Client/src/app/data-sets/shared/data-set.model.ts
export class DataSetModel {
    public id: number;
    public name: string;
    public description: string;
    public insertDate: Date;
    public deleteDate: Date;
    public logo: string;
    public userId: number;
    public fileEntries: any[];

    public constructor(fields?: Partial<DataSetModel>) {

        if (fields) {
            Object.assign(this, fields);
        }
    }
}
